### ISSUE TITLE
Blur foreground?

### ISSUE THREAD

**Issue by rajansaini691** [yshui/picom#489]
First of all, thanks for maintaining this repo, it's awesome!

Is it possible to blur the foreground of inactive windows? Kind of like typora's focus mode? If not, I'd be happy to add a PR, just let me know what would be least intrusive.

**Comment by mighty9245** [comment]
no, it's not possible to blur the foreground of inactive windows now

you can try to achieve *something* like this by using inversion in opacity rules and blur exclusions to give all inactive windows opacity and blur, idk how it'll look tho

**Comment by yshui** [comment]
@rajansaini691 I would be happy to review such PR. If you are going to do it, please implement it for the experimental backends. I think it should be pretty easily achievable.

**Comment by rajansaini691** [comment]
@mighty9245 Thanks. I'll probably just directly implement it, then. 

@yshui Awesome! Which branch should I use? Is it the next branch? Also, I'm trying to think of a good function name; how does `blur_fg` sound? Should the current `blur` be left as-is, or should it be refactored to `blur_bg`?

**Comment by tryone144** [comment]
@rajansaini691 `next` is the current development head.

I think it should be possible to reuse most of the current implementation for blurring as is and just change the sampling source. No need to re-implement the actual blurring magic. :wink: 
Actual window rendering in the experimental backends is done in `render_all_new()` in [backend.c](https://github.com/yshui/picom/blob/next/src/backend/backend.c). You can find the different compositing steps there as a starting point.
